eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/ChannelOpenDualFunded.scala

```diff
@@ -362,23 +363,156 @@ trait ChannelOpenDualFunded extends DualFundingHandlers with ErrorHandlers {
             stay() using d1 storing() calling publishFundingTx(d.fundingParams, fundingTx)
         }
         case _: FullySignedSharedTransaction =>
-          log.warning("received duplicate tx_signatures")
-          stay()
+          d.rbfAttempt match {
+            case Some(Right(txBuilder)) =>
+              txBuilder ! InteractiveTxBuilder.ReceiveTxSigs(txSigs)
+              stay()
+            case _ =>
+              // Signatures are retransmitted on reconnection, but we may have already received them.
+              log.info("ignoring duplicate tx_signatures for txid={}", txSigs.txId)
+              stay()
+          }
       }
 
-    case Event(_: TxInitRbf, _: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) =>
-      log.info("rbf not supported yet")
-      stay()
+    case Event(msg: TxInitRbf, d: DATA_WAIT_FOR_DUAL_FUNDING_CONFIRMED) =>
+      if (d.fundingParams.isInitiator) {
+        // Only the initiator is allowed to initiate RBF.
+        log.info("rejecting tx_init_rbf, we're the initiator, not them!")
+        stay() sending TxAbort(d.channelId, InvalidRbfAttempt(d.channelId).getMessage)
+      } else {
+        val minNextFeerate = d.fundingParams.targetFeerate * 25 / 24
```

Comment:
This ratio should probably be factored.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2275/commits/ba3baefe8114d5930c8049d530a2224e1b8b104b